lib/portage/package/ebuild/config.py

```diff
@@ -1675,13 +1675,10 @@ def setcpv(self, mycpv, use_cache=None, mydb=None):
 					restrict = use_reduce(raw_restrict,
 						uselist=built_use, flat=True)
 				else:
-					# Use matchnone=True to ignore USE conditional parts
-					# of RESTRICT, since we want to know whether to mask
-					# the "test" flag _before_ we know the USE values
-					# that would be needed to evaluate the USE
-					# conditionals (see bug #273272).
 					restrict = use_reduce(raw_restrict,
-						matchnone=True, flat=True)
+						uselist=frozenset(x for x in self['USE'].split()
```

Comment:
Hmm, I've just realize that a `self.configdict["features"]["USE"]` setting from an earlier package could corrupt `self['USE']` here...

Reply:
I've solved this by making `self.configdict["features"]["USE"]` independent of IUSE, and by resetting `self.configdict["features"]["USE"]` when the previous package had a FEATURES setting from package.env.

Reply:
However, `self.configdict["features"]["USE"]` is still dependent on RESTRICT, so I need to eliminate that dependency or handle it somehow.

Reply:
I've solve this by making features USE independent of RESTRICT. When the evaluated value of RESTRICT contains "test", it's now handled like IUSE="-test".